src/NServiceBus.Core/Routing/AssemblyRouteSource.cs

```diff
@@ -0,0 +1,28 @@
+namespace NServiceBus
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using Routing;
+
+    class AssemblyRouteSource : IRouteSource
+    {
+        Assembly messageAssembly;
+        Conventions conventions;
+        IUnicastRoute route;
+
+        public AssemblyRouteSource(Assembly messageAssembly, Conventions conventions, IUnicastRoute route)
+        {
+            this.messageAssembly = messageAssembly;
+            this.conventions = conventions;
```

Comment:
I think it's dangerous to pass the conventions on the ctor. It may let you think it's ok to use the conventions at this point. If someone changes the conventions after registering routes, this will break a lot of things. Here it will work because we use the conventions only during feature setup and because we only have a single instance. Still I think we should probably just pass the conventions into `GenerateRoutes` as parameter.